waspc/data/Generator/templates/sdk/wasp/server/config.ts

```diff
@@ -40,7 +41,7 @@ const config: {
   all: {
     env,
     isDevelopment: env === 'development',
-    port: parseInt(process.env.PORT) || 3001,
+    port: parseInt(process.env.PORT) || {= defaultServerPort =},
```

Comment:
@sodic Now the default server port and default server URL are both defined in Haskell and the URL is constructed using the default port variable.